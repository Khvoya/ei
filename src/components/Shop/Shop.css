#shop {
  padding-bottom: 4vw;
}
.Shop__title {
  margin: 4vw 0;
}
.Shop__slider-arrow {
  border: none;
  background-color: unset;
  position: absolute;
  top: 20vh;
  margin: 0;
  z-index: 1;
}
.Shop__slider-arrow-left {
  left: -60px;
}
.Shop__slider-arrow-right {
  right: -60px;
  transform: rotate(180deg);
}

.Shop__slider-arrow:focus,
.Shop__slider-arrow:active {
  outline: none;
  box-shadow: none;
}
.Shop__button-wrap {
  width: 100%;
  position: relative;
}

.btn-primary,
.btn-primary:focus,
.btn-primary:focus-within,
.btn-primary:active,
.btn-primary:hover,
.btn-primary:visited {
  color: black;
  background-color: white;
  border-color: white;
  box-shadow: none;
  outline: none;
  cursor: pointer;
}
.btn-primary:not(:disabled):not(.disabled).active,
.btn-primary:not(:disabled):not(.disabled):active,
.show > .btn-primary.dropdown-toggle {
  color: black;
  background-color: white;
  border-color: white;
  outline: none;
  box-shadow: none;
}
@media screen and (max-width: 767px) {
  .Shop__slider-arrow {
    margin: 50px 30px;
  }
}

@media screen and (max-width: 400px) {
  .Shop__slider-arrow {
    display: none;
  }
}
