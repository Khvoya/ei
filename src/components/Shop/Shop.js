import React, { Component } from "react";
import "./Shop.css";
import {Container, Col, Row} from 'react-bootstrap';
import leftArrow from '../../media/icons/arrow-pointing-to-left.svg'
import rightArrow from '../../media/icons/arrow-pointing-to-right.svg'
import TinySlider from "tiny-slider-react";
import { SHOP_ITEMS } from "../../constants/shop";
import ShopItem from "../../components/Shop/ShopItem/ShopItem";

class Shop extends Component {
  onGoTo = dir => this.ts.slider.goTo(dir)

  render() {
    const settings = {
      lazyload: true,
      nav: false,
      mouseDrag: true,
      autoWidth: true,
      speed: 500,
      controls: false,
      responsive: {
        1200: {
          items: 3,
        },
        900: {
          items: 2,
        }
      },
    };
    const items = SHOP_ITEMS.map(item => (
      <ShopItem
        src={item.image}
        name={item.name}
        price={item.price}
        key={item.name}
      />
    ));
    return (
      <Container id="shop" className="align-content-center justify-content-around">
        <Row>
          <Col>
            <h1 className="Shop__title">Shop</h1>
          </Col>
        </Row>

          <TinySlider settings={settings} ref={ts => this.ts = ts}>
            {items}
          </TinySlider>
        <button className="Shop__slider-arrow" type="button" onClick={() => this.onGoTo('prev')}>
          <img src={leftArrow} width="50px" height="50px"/>
        </button>
        <button className="Shop__slider-arrow" type="button" onClick={() =>  this.onGoTo('next')}>
          <img src={rightArrow} width="50px" height="50px"/>
        </button>
      </Container>
    );
  }
}

export default Shop;
