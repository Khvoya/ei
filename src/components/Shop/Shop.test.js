import React from "react";
import { shallow } from "enzyme";
import { Shop } from "./Shop.js";

jest.mock('tiny-slider-react', () => 'TinySlider');

describe("Shop", () => {

  const props = {
    shop: [
      {
        description: "test description",
        id: "23",
        imageUrl: ["https://test-image1.url", "https://test-image2.url"],
        material: "material",
        name: "test name",
        price: "99",
        size: "xl"
      }
    ]
  };

  it("should render correctly", () => {
    const output = shallow(<Shop {...props} />);
    expect(output).toMatchSnapshot();
  });
});
