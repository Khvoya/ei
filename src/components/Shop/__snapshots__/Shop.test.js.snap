// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Shop
    shop={
      Array [
        Object {
          "description": "test description",
          "id": "23",
          "imageUrl": Array [
            "https://test-image1.url",
            "https://test-image2.url",
          ],
          "material": "material",
          "name": "test name",
          "price": "99",
          "size": "xl",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": Array [
        <ForwardRef(render)>
          <ForwardRef
            as="div"
          >
            <h1
              className="Shop__title"
            >
              Shop
            </h1>
          </ForwardRef>
        </ForwardRef(render)>,
        <div
          className="Shop__button-wrap"
        >
          <button
            className="Shop__slider-arrow Shop__slider-arrow-left"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="left-arrow"
              height="50px"
              src="arrow-pointing-to-left.svg"
              width="50px"
            />
          </button>
          <button
            className="Shop__slider-arrow Shop__slider-arrow-right"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="right-arrow"
              height="50px"
              src="arrow-pointing-to-left.svg"
              width="50px"
            />
          </button>
          <TinySlider
            settings={
              Object {
                "arrowKeys": true,
                "autoHeight": true,
                "controls": false,
                "gutter": 20,
                "lazyload": true,
                "loop": false,
                "mouseDrag": true,
                "nav": false,
                "responsive": Object {
                  "0": Object {
                    "items": 1,
                  },
                  "1200": Object {
                    "items": 3,
                  },
                  "1600": Object {
                    "items": 4,
                  },
                  "800": Object {
                    "autoWidth": true,
                    "items": 2,
                  },
                },
                "rewind": true,
                "speed": 500,
              }
            }
          >
            <ShopItem
              description="test description"
              id="23"
              imageUrl={
                Array [
                  "https://test-image1.url",
                  "https://test-image2.url",
                ]
              }
              material="material"
              name="test name"
              price="99"
              size="xl"
            />
          </TinySlider>
        </div>,
      ],
      "className": "align-content-center justify-content-around",
      "fluid": false,
      "id": "shop",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
          >
            <h1
              className="Shop__title"
            >
              Shop
            </h1>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <h1
              className="Shop__title"
            >
              Shop
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Shop",
              "className": "Shop__title",
            },
            "ref": null,
            "rendered": "Shop",
            "type": "h1",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="Shop__slider-arrow Shop__slider-arrow-left"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="left-arrow"
                height="50px"
                src="arrow-pointing-to-left.svg"
                width="50px"
              />
            </button>,
            <button
              className="Shop__slider-arrow Shop__slider-arrow-right"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="right-arrow"
                height="50px"
                src="arrow-pointing-to-left.svg"
                width="50px"
              />
            </button>,
            <TinySlider
              settings={
                Object {
                  "arrowKeys": true,
                  "autoHeight": true,
                  "controls": false,
                  "gutter": 20,
                  "lazyload": true,
                  "loop": false,
                  "mouseDrag": true,
                  "nav": false,
                  "responsive": Object {
                    "0": Object {
                      "items": 1,
                    },
                    "1200": Object {
                      "items": 3,
                    },
                    "1600": Object {
                      "items": 4,
                    },
                    "800": Object {
                      "autoWidth": true,
                      "items": 2,
                    },
                  },
                  "rewind": true,
                  "speed": 500,
                }
              }
            >
              <ShopItem
                description="test description"
                id="23"
                imageUrl={
                  Array [
                    "https://test-image1.url",
                    "https://test-image2.url",
                  ]
                }
                material="material"
                name="test name"
                price="99"
                size="xl"
              />
            </TinySlider>,
          ],
          "className": "Shop__button-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="left-arrow"
                height="50px"
                src="arrow-pointing-to-left.svg"
                width="50px"
              />,
              "className": "Shop__slider-arrow Shop__slider-arrow-left",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left-arrow",
                "height": "50px",
                "src": "arrow-pointing-to-left.svg",
                "width": "50px",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="right-arrow"
                height="50px"
                src="arrow-pointing-to-left.svg"
                width="50px"
              />,
              "className": "Shop__slider-arrow Shop__slider-arrow-right",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right-arrow",
                "height": "50px",
                "src": "arrow-pointing-to-left.svg",
                "width": "50px",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ShopItem
                  description="test description"
                  id="23"
                  imageUrl={
                    Array [
                      "https://test-image1.url",
                      "https://test-image2.url",
                    ]
                  }
                  material="material"
                  name="test name"
                  price="99"
                  size="xl"
                />,
              ],
              "settings": Object {
                "arrowKeys": true,
                "autoHeight": true,
                "controls": false,
                "gutter": 20,
                "lazyload": true,
                "loop": false,
                "mouseDrag": true,
                "nav": false,
                "responsive": Object {
                  "0": Object {
                    "items": 1,
                  },
                  "1200": Object {
                    "items": 3,
                  },
                  "1600": Object {
                    "items": 4,
                  },
                  "800": Object {
                    "autoWidth": true,
                    "items": 2,
                  },
                },
                "rewind": true,
                "speed": 500,
              },
            },
            "ref": [Function],
            "rendered": Array [
              Object {
                "instance": null,
                "key": "23",
                "nodeType": "class",
                "props": Object {
                  "description": "test description",
                  "id": "23",
                  "imageUrl": Array [
                    "https://test-image1.url",
                    "https://test-image2.url",
                  ],
                  "material": "material",
                  "name": "test name",
                  "price": "99",
                  "size": "xl",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "TinySlider",
          },
        ],
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": Array [
          <ForwardRef(render)>
            <ForwardRef
              as="div"
            >
              <h1
                className="Shop__title"
              >
                Shop
              </h1>
            </ForwardRef>
          </ForwardRef(render)>,
          <div
            className="Shop__button-wrap"
          >
            <button
              className="Shop__slider-arrow Shop__slider-arrow-left"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="left-arrow"
                height="50px"
                src="arrow-pointing-to-left.svg"
                width="50px"
              />
            </button>
            <button
              className="Shop__slider-arrow Shop__slider-arrow-right"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="right-arrow"
                height="50px"
                src="arrow-pointing-to-left.svg"
                width="50px"
              />
            </button>
            <TinySlider
              settings={
                Object {
                  "arrowKeys": true,
                  "autoHeight": true,
                  "controls": false,
                  "gutter": 20,
                  "lazyload": true,
                  "loop": false,
                  "mouseDrag": true,
                  "nav": false,
                  "responsive": Object {
                    "0": Object {
                      "items": 1,
                    },
                    "1200": Object {
                      "items": 3,
                    },
                    "1600": Object {
                      "items": 4,
                    },
                    "800": Object {
                      "autoWidth": true,
                      "items": 2,
                    },
                  },
                  "rewind": true,
                  "speed": 500,
                }
              }
            >
              <ShopItem
                description="test description"
                id="23"
                imageUrl={
                  Array [
                    "https://test-image1.url",
                    "https://test-image2.url",
                  ]
                }
                material="material"
                name="test name"
                price="99"
                size="xl"
              />
            </TinySlider>
          </div>,
        ],
        "className": "align-content-center justify-content-around",
        "fluid": false,
        "id": "shop",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              as="div"
            >
              <h1
                className="Shop__title"
              >
                Shop
              </h1>
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <h1
                className="Shop__title"
              >
                Shop
              </h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Shop",
                "className": "Shop__title",
              },
              "ref": null,
              "rendered": "Shop",
              "type": "h1",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="Shop__slider-arrow Shop__slider-arrow-left"
                onClick={[Function]}
                type="button"
              >
                <img
                  alt="left-arrow"
                  height="50px"
                  src="arrow-pointing-to-left.svg"
                  width="50px"
                />
              </button>,
              <button
                className="Shop__slider-arrow Shop__slider-arrow-right"
                onClick={[Function]}
                type="button"
              >
                <img
                  alt="right-arrow"
                  height="50px"
                  src="arrow-pointing-to-left.svg"
                  width="50px"
                />
              </button>,
              <TinySlider
                settings={
                  Object {
                    "arrowKeys": true,
                    "autoHeight": true,
                    "controls": false,
                    "gutter": 20,
                    "lazyload": true,
                    "loop": false,
                    "mouseDrag": true,
                    "nav": false,
                    "responsive": Object {
                      "0": Object {
                        "items": 1,
                      },
                      "1200": Object {
                        "items": 3,
                      },
                      "1600": Object {
                        "items": 4,
                      },
                      "800": Object {
                        "autoWidth": true,
                        "items": 2,
                      },
                    },
                    "rewind": true,
                    "speed": 500,
                  }
                }
              >
                <ShopItem
                  description="test description"
                  id="23"
                  imageUrl={
                    Array [
                      "https://test-image1.url",
                      "https://test-image2.url",
                    ]
                  }
                  material="material"
                  name="test name"
                  price="99"
                  size="xl"
                />
              </TinySlider>,
            ],
            "className": "Shop__button-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="left-arrow"
                  height="50px"
                  src="arrow-pointing-to-left.svg"
                  width="50px"
                />,
                "className": "Shop__slider-arrow Shop__slider-arrow-left",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "left-arrow",
                  "height": "50px",
                  "src": "arrow-pointing-to-left.svg",
                  "width": "50px",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="right-arrow"
                  height="50px"
                  src="arrow-pointing-to-left.svg"
                  width="50px"
                />,
                "className": "Shop__slider-arrow Shop__slider-arrow-right",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "right-arrow",
                  "height": "50px",
                  "src": "arrow-pointing-to-left.svg",
                  "width": "50px",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ShopItem
                    description="test description"
                    id="23"
                    imageUrl={
                      Array [
                        "https://test-image1.url",
                        "https://test-image2.url",
                      ]
                    }
                    material="material"
                    name="test name"
                    price="99"
                    size="xl"
                  />,
                ],
                "settings": Object {
                  "arrowKeys": true,
                  "autoHeight": true,
                  "controls": false,
                  "gutter": 20,
                  "lazyload": true,
                  "loop": false,
                  "mouseDrag": true,
                  "nav": false,
                  "responsive": Object {
                    "0": Object {
                      "items": 1,
                    },
                    "1200": Object {
                      "items": 3,
                    },
                    "1600": Object {
                      "items": 4,
                    },
                    "800": Object {
                      "autoWidth": true,
                      "items": 2,
                    },
                  },
                  "rewind": true,
                  "speed": 500,
                },
              },
              "ref": [Function],
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "23",
                  "nodeType": "class",
                  "props": Object {
                    "description": "test description",
                    "id": "23",
                    "imageUrl": Array [
                      "https://test-image1.url",
                      "https://test-image2.url",
                    ],
                    "material": "material",
                    "name": "test name",
                    "price": "99",
                    "size": "xl",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "TinySlider",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
